src/components/users/add-member-dialog.tsx

```diff
@@ -15,120 +19,277 @@ interface Props {
-          >
-            <MenuItem value={ MetaformMemberRole.Administrator }>{ strings.userManagementScreen.addMemberDialog.roleAdministrator }</MenuItem>
-            <MenuItem value={ MetaformMemberRole.Manager }>{ strings.userManagementScreen.addMemberDialog.roleManager }</MenuItem>
-          </Select>
-        </Stack>
-      </DialogContent>
-      <DialogActions style={{ padding: 24, paddingTop: 0 }}>
-        <Button disableElevation variant="contained" onClick={ onCancel } color="secondary" autoFocus>
-          { strings.userManagementScreen.addMemberDialog.cancelButton }
+        </Tooltip>
+        <Button
+          sx={{ flex: 0.25, height: 56 }}
+          size="large"
+          onClick={ handleSearchButtonClick }
+          disabled={ loading }
+        >
+          { strings.userManagementScreen.addMemberDialog.searchButton }
         </Button>
-        <GenericLoaderWrapper loading={ loading }>
-          <Button onClick={ onCreateClick } color="primary" disabled={ !valid }>
-            { strings.userManagementScreen.addMemberDialog.createButton }
-          </Button>
-        </GenericLoaderWrapper>
-      </DialogActions>
-    </Dialog>
+      </Stack>
+      <TextField
+        select
+        fullWidth
+        size="medium"
+        onChange={ handleSelectChange }
+        label={ strings.userManagementScreen.addMemberDialog.usersSelectLabel }
+      >
+        { foundUsers.map(renderSelectableUsers) }
+      </TextField>
+      <TextField
+        disabled={ !selectedUser ?? loading }
+        fullWidth
+        size="medium"
+        required
+        value={ selectedUser?.email ?? "" }
+        type="email"
+        name="email"
+        label={ strings.userManagementScreen.addMemberDialog.emailLabel }
+        onChange={ handleTextFieldChange }
+      />
+      <TextField
+        disabled={ !selectedUser ?? loading }
+        fullWidth
+        size="medium"
+        required
+        value={ selectedUser?.firstName ?? "" }
+        name="firstName"
+        label={ strings.userManagementScreen.addMemberDialog.firstNameLabel }
+        onChange={ handleTextFieldChange }
+      />
+      <TextField
+        disabled={ !selectedUser ?? loading }
+        fullWidth
+        size="medium"
+        required
+        name="lastName"
+        value={ selectedUser?.lastName ?? "" }
+        label={ strings.userManagementScreen.addMemberDialog.lastNameLabel }
+        onChange={ handleTextFieldChange }
+      />
+      <TextField
+        disabled
+        fullWidth
+        size="medium"
+        value={ (selectedUser && getUsersUpnNumber(selectedUser)) ?? "" }
+        label={ strings.userManagementScreen.addMemberDialog.upnNumberLabel }
+        onChange={ handleTextFieldChange }
+      />
+    </Stack>
+  );
+
+  /**
+   * Renders dialog actions
+   */
+  const renderDialogActions = () => [
+    <Button disableElevation variant="contained" onClick={ handleCancelClick } color="secondary" autoFocus>
```

Comment:
Should cancel button also be with loader, does it actually "cancel" anything if api call is made?

Reply:
Resolved